CONTRIBUTING.md

```diff
@@ -0,0 +1,176 @@
+
+We assume you use Linux as a work environment (you can use a virtual machine as well).
+
+1. Install ``docker``, launch it.
+
+2. Clone the [ansible-core](https://github.com/ansible/ansible) repository:
+```bash
+git clone https://github.com/ansible/ansible.git
+```
+3. Go to the cloned repository and prepare the environment:
+```bash
+cd ansible && source hacking/env-setup
+pip install -r requirements.txt
+cd ~
+```
+4. Create the following directories in your home directory:
+```bash
+mkdir -p ~/ansible_collections/community/mysql
+```
+
+5. Fork the ``community.mysql`` repository via the GitHub web interface.
+
+6. Clone the forked repository from your profile to the created path:
+```bash
+git clone https://github.com/YOURACC/community.mysql.git ~/ansible_collections/community/mysql
+```
+
+7. Go there:
+```bash
+cd ~/ansible_collections/community/mysql
+```
+
+8. Be sure you are in the main branch:
+```bash
+git status
+```
+
+9. Show remotes. There should be the ``origin`` repository only:
+```bash
+git remote -v
+```
+
+10. Add the ``upstream`` repository:
+```bash
+git remote add upstream https://github.com/ansible-collections/community.mysql.git
+```
+
+11. Update your local ``main`` branch:
+```bash
+git fetch upstream
+git merge upstream/main
+```
+
+12. Create a branch for your changes:
+```bash
+git branch -b name_of_my_branch
+```
+
+13. It can be a good approach is to start with writing integration tests if applicable.
+
+Note: If there are any difficulties with writing the tests or you are not sure if the case can be covered, feel free to skip this step.
+If needed, other contributors can help you with it later.
+
+All integration tests are stored in ``tests/integration/targets`` subdirectories.
+You are interested in a subdirectory containing a name of module you are going to change.
+For example, if you are fixing the ``mysql_user`` module, its tests are in ``tests/integration/targets/test_mysql_user/tasks``
+
+There is the ``main.yml`` file which includes other test files.
+Look for a suitable one to integrate your tests or create and include a dedicated one.
+You can use one of the existing as a draft.
+
+When fixing a bug, write a task which reproduces the bug from the issue.
+
+Put the reported case in the tests, then run integration tests with the following command:
+```bash
+ansible-test integration test_mysql_user --docker -vvv > ~/test.log
+```
+If the tests do not want to run, first, check you complete step 3 of this guide.
+
+If the tests ran successfully, there are usually two possible outcomes:
+a) If the bug has not appeared and the tests have passed successfully, ask the reporter to provide more details. The bug can be not a bug actually or can relate to a particular software version used or specifics of local environment configuration.
+
+b) The bug has appeared and the tests has failed as expected showing the reported symptoms.
+
+14. Fix the bug.
+
+15. Run ``flake8``:
```

Comment:
I have never used `flake8`, actually never heard of it, probably because I'm not a python developer :).

So do you consider this much useful to be included in this documentation?

I have not used `flake8` again, so I'm no sure.

Reply:
It helped me much. It runs very quickly - almost instant, and helps find most common issues like wrong indentation, spaces, etc. + unused imports which are not shown by sanity tests.
I think it's worth adding it (but it's only my opinion)

Reply:
I'm using it a lot as well, it has the advantage over pylint, ansible-test and other tools that it is so fast. The only annoying part is the line length limit, but then there's `--ignore=E501` which is already part of my muscle memory ;-)

Reply:
Do we need to include `--max-line-length=160`? 
I haven't used `flake8` so not sure what's needed

Reply:
@gundalow Thanks, I didn't know it exists:) My personal preference is to see warnings generated by flake8 because code looks much better if the lines are around 80-100. In case of 160, the tool will be silent. Let's see what other folks think.
@felixfontein @aminvakil your thoughts?

Reply:
I can only think of adding `(optional)` or something like that to this step so that if user wanted to stick to `ansible-test` doesn't feel they have left out a step. Other than that I agree with @Andersson007 about seeing a warning when line is too lengthy.

Reply:
Sounds sensible, thanks

Reply:
I mentioned the argument and that this is optional